src/app/sessions/services/sessions-service.ts

```diff
@@ -16,10 +16,16 @@ export class SessionsService {
     constructor(private http: HttpClient, private config: AppConfig) {
     }
 
+    getUserTransaction(id: string | String): Observable<any> {
```

Comment:
getSessionSByUserTransactionId ?

Reply:
This method gets only the user transaction. The casting to <Session> is misleading and should not be there.